sdk/src/system_transaction.rs

```diff
@@ -30,7 +27,11 @@ pub fn create_account(
 }
 
 /// Create and sign new system_instruction::Assign transaction
-pub fn assign(from_keypair: &Keypair, recent_blockhash: Hash, program_id: &Pubkey) -> Transaction {
+pub fn assign(
+    from_keypair: &dyn KeypairUtil,
```

Comment:
This makes it easier to create bad transactions. Anything which takes a `KeypairUtil` could create bad pubkeys or bad signatures because of the fallback behavior in `KeypairUtil::pubkey()` and `KeypairUtil::sign_message`.  I'm not sure it's a good idea to make this easier. Anyone who creates a system transaction with a remote keypair or presigner keypair will have to be aware that the transaction they create, might fail due to an error that isn't bubbled up.

Reply:
I think we can eliminate the default fallbacks. `sign_message()` has a manageable number of callers, many just setting up tests, so probably not too bad to make fallible throughout. `pubkey()` however, is called all over the place.  I believe it can be made infallible, though. Any implementers that might fail, can query and cache the pubkey at construction and fail there instead.  @CriesofCarrots wdyt?

@jstarry would that alleviate your concerns RE default fallbacks?